### PR TITLE
Add test for db migrations

### PR DESCRIPTION
**Pull request by jhrozek**
- make: Add a target to download the migrate binary

Adds a `make migrate` target for the Makefile that fetches
the migrate CLI. Adds several top-level Makefile variables for
autodetection of the OS and the arch so that we support different OS-es
and arches from the start.

- tests: Add a GH action test for the DB migrations

Adds a github action that spawns a postgres container and runs migrate
up and migrate down to exercise our database migrations.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+
+.PHONY: build
+build: $(BUILDS_DIR)
 	go build -o builds/ cmd/compserv-server.go
+
+.PHONY: $(BUILDS_DIR)
+$(BUILDS_DIR):
+	mkdir -p $(BUILDS_DIR)
+
+.PHONY: $(BUILD_DIR)/migrate
```
Comment (jhrozek) [inline Makefile:16]: I wasn't sure if it's OK to use the same dir for the helper utilities as we do for the main project binaries. But we can use whatever directory, maybe a subdir of builds like `builds/tools` ?

File: .github/workflows/migrations.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Install golang-migrate
+      run: |
+        mkdir -p ${GITHUB_WORKSPACE}/builds
+        echo "${GITHUB_WORKSPACE}/builds" >> ${GITHUB_PATH}
+        make builds/migrate
+
+    - name: Run migrations
+      run: |
+        POSTGRESQL_URL="postgres://$DB_USER:$DB_PASSWORD@localhost:5432/$DB_NAME?sslmode=disable"
```
Comment (jhrozek) [inline .github/workflows/migrations.yml:44]: Is it expected that the test will ever be more complex? If yes, we might want to split the test into a shell of its own so it's easier to run it locally. Same with spinning up the container, I used the gh actions' built-in `services', but I guess it would be possible to spawn the container from a helper script and then the script would be runnable outside the context of actions..

File: Makefile (hunk 1 above)
Comment (jhrozek) [reply Makefile:16]: thank you for the feedback, I'll change this

File: .github/workflows/migrations.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: migrations
```
Comment (rhmdnd) [inline .github/workflows/migrations.yml:1]: nit: We could make this more descriptive with something like `PostgreSQL Migration CI`.

Also, I used the canned actions for Go and Make, which name the actions using upper-case conventions (Go/Makefile CI). We could use that convention here, too.

File: Makefile (hunk 1 above)
Comment (jhrozek) [reply Makefile]: Done

File: .github/workflows/migrations.yml (hunk 2 above)
Comment (jhrozek) [reply .github/workflows/migrations.yml]: Done, please see the new script and the readme blob under migrations that describes how to run it (spolier: it's a one-liner)

File: .github/workflows/migrations.yml (hunk 3 above)
Comment (jhrozek) [reply .github/workflows/migrations.yml]: thanks, fixed

File: migrations/test.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+$RUNTIME run -d --name postgres \
+    -e POSTGRES_USER=$DB_USER \
+    -e POSTGRES_DB=$DB_NAME \
+    -e POSTGRES_PASSWORD=$DB_PASSWORD \
+    -p 5432:5432 \
+    --health-cmd pg_isready \
+    --health-interval 10s \
+    --health-timeout 5s \
+    --health-retries 5 \
+    postgres:latest
```
Comment (rhmdnd) [inline migrations/test.sh:55]: We can think about how we want to do this in another patch,  but I think parameterizing the postgres version would be a good idea.

I know the `terraform` and RDS bits we have spin up postgres 13.4. In theory, we could hit an issue where migrations work in one place, but not another due to the postgres versions.

File: migrations/test.sh [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        sleep 3
+    done
+
+    if [ $health_status != "healthy" ] ; then
+        echo "Failed to wait for pgsql container to come up"
+        exit 1
+    fi
+}
+
+$RUNTIME run -d --name postgres \
```
Comment (rhmdnd) [inline migrations/test.sh:46]: Here we keep the container around for debug purposes? I suppose for CI is doesn't matter if we `--rm` the container or not.

File: migrations/test.sh [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    --health-timeout 5s \
+    --health-retries 5 \
+    postgres:latest
+
+wait_for_db_init
+
+POSTGRESQL_URL="postgres://$DB_USER:$DB_PASSWORD@$DB_HOST:5432/$DB_NAME?sslmode=disable"
+
+migrate -database $POSTGRESQL_URL -path migrations up
+echo "y" | migrate -database $POSTGRESQL_URL -path migrations down
```
Comment (rhmdnd) [inline migrations/test.sh:62]: Another thing we can think about and fix in a subsequent patch if we choose.

I wonder if we should decouple this further and separate the DB setup from the test invocation. I'm thinking about the perspective where I might want to migrate up to N, and inspect the database schema, then step through other migrations like N+1, N+2.

**Review by rhmdnd — APPROVED** [review]
